FunctionalTableData/Separator.swift

```diff
@@ -14,34 +14,38 @@ import UIKit
 ///
 /// Supported by `UITableView` only.
 public class Separator: UIView {
-	/// Specifies the default inset of cell separators.
-	public static var inset: CGFloat = 0.0
-	/// Specifies the default thickness of cell separators.
-	private let thickness: CGFloat = 1.0 / UIScreen.main.scale
-	
 	/// The style for table cells used as separators.
-	///
-	/// The options are `full`, `inset`, `moreInset`, `custom`
-	public enum Style: Equatable {
-		case full
-		case inset
-		case moreInset
-		case custom(leadingInset: CGFloat, trailingInset: CGFloat, layoutMarginsRelative: Bool)
-		
-		public static func ==(lhs: Style, rhs: Style) -> Bool {
-			switch (lhs, rhs) {
-			case let (.custom(lhsLeadingInset, lhsTrailingInset, lhsLayoutMargins), .custom(rhsLeadingInset, rhsTrailingInset, rhsLayoutMargins)):
-				return lhsLeadingInset == rhsLeadingInset && lhsTrailingInset == rhsTrailingInset && lhsLayoutMargins == rhsLayoutMargins
-			case (.full, .full):
-				return true
-			case (.inset, .inset):
-				return true
-			case (.moreInset, .moreInset):
-				return true
-			default:
-				return false
+	public struct Style: Equatable {
+		/// The inset used in the separators.
+		public struct Inset: Equatable {
+			/// Specifies the amount of spacing to apply to the separator
+			public let value: CGFloat
+			/// Specifies if the inset is relative to the layout margins.
+			public let respectingLayoutMargins: Bool
+			
+			public static let none: Inset = Inset(value: 0, respectingLayoutMargins: false)
+			
+			public init(value: CGFloat, respectingLayoutMargins: Bool) {
+				self.value = value
+				self.respectingLayoutMargins = respectingLayoutMargins
 			}
 		}
+		
+		/// Specifies the leading inset of the separators.
+		public let leadingInset: Inset
+		/// Specifies the trailing inset of the separators.
+		public let trailingInset: Inset
+		/// Specifies the thickness of cell separators.
+		public let thickness: CGFloat
+		
+		static public let full: Style = Style(leadingInset: .none, trailingInset: .none)
+		static public let inset: Style = Style(leadingInset: .init(value: 0, respectingLayoutMargins: true), trailingInset: .none)
+		
+		public init(leadingInset: Inset, trailingInset: Inset, thickness: CGFloat = 1.0) {
+			self.leadingInset = leadingInset
+			self.trailingInset = trailingInset
+			self.thickness = thickness / UIScreen.main.scale
```

Comment:
Too magic? people won't know this is happening. Tripped me for a second in the tests